src/zip.c

```diff
@@ -870,6 +870,14 @@ int zip_is64(struct zip_t *zip) {
 }
 
 int zip_entry_open(struct zip_t *zip, const char *entryname) {
+  if (!zip) {
+    return ZIP_ENOINIT;
+  }
+
+  if (!entryname) {
+    return ZIP_EINVENTNAME;
+  }
+
```

Comment:
I would not move this part to the top, just leave declarations first (to follow c standards)